### PR TITLE
Add authentication with Keycloak for scraping

### PR DESCRIPTION
**Pull request by spahrson**
This PR add the ability to authenticate by Keycloak.

### CODE REVIEW

**Comment by curquiza** [comment]
Hello @spahrson!
Thanks for this PR, it seems to be a great addition! I don't know keycloak, can you tell me a little bit about it? What is your usage of keycloak for example?
I will discuss about it with the team and I will come back to you before the end of the week 🙂 But it should be ok!

**Comment by spahrson** [comment]
Hi @curquiza.

Keycloak is a open source identity and access management solution. It can act as an OAuth 2.0 and OpenID connect compliant authorization server and there is a proxy called Gatekeeper to protect non-oauth compliant applications. We use Keycloak as Single-Sign-on solution. We use antora for documentation and meilisearch to search our documentation. Our antora instances are protected by Keycloak. So we need the ability to scrape antora for meilisearch using an OAuth 2.0 Bearer-Token. This PR fullfills this requirement for Keycloak.

File: README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ This scraper is used in production and runs on the [MeiliSearch documentation](h
     - [`min_indexed_level` (optional)](#min_indexed_level-optional)
     - [`only_content_level` (optional)](#only_content_level-optional)
   - [Authentication](#authentication)
+    - [Keycloak Access: Identity and Access Management](#keycloak-access-identity-and-access-management)
```
Comment (curquiza) [inline README.md:53]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -480,6 +481,12 @@ In case of Google Cloud Identity-Aware Proxy, please specify these env variables
 - `IAP_AUTH_CLIENT_ID` - # pick [client ID of the application](https://console.cloud.google.com/apis/credentials) you are connecting to
 - `IAP_AUTH_SERVICE_ACCOUNT_JSON` - # generate in [Actions](https://console.cloud.google.com/iam-admin/serviceaccounts) -> Create key -> JSON
 
+#### Keycloak Access: Identity and Access Management
+
+If you need to scrape site protected by Keycloak (Gatekeeper), you have to provide a valid Access Token.
```
Comment (curquiza) [inline README.md:486]:
```suggestion
If you need to scrape site protected by [Keycloak](https://github.com/keycloak/keycloak) (Gatekeeper), you have to provide a valid access token.
```

File: README.md [hunk 3]
```diff
@@ -480,6 +481,12 @@ In case of Google Cloud Identity-Aware Proxy, please specify these env variables
...
+
+If you set the env variables `KC_URL`, `KC_REALM`, `KC_CLIENT_ID` and `KC_CLIENT_SECRET` the scraper authenticates himself against Keycloak using _Client Credentials Grant_ and adds the resulting Access Token as `Authorization` HTTP header to each scraping request.
```
Comment (curquiza) [inline README.md:488]:
```suggestion
If you set the environment variables `KC_URL`, `KC_REALM`, `KC_CLIENT_ID`, and `KC_CLIENT_SECRET` the scraper authenticates itself against Keycloak using _Client Credentials Grant_ and adds the resulting access token as `Authorization` HTTP header to each scraping request.
```

File: scraper/src/index.py [hunk 4]
```diff
@@ -114,3 +125,4 @@ def run_config(config):
     from os import environ
 
     run_config(environ['CONFIG'])
+
```
Comment (curquiza) [inline scraper/src/index.py:128]: Could you remove the empty space in this line?
Plus the file should end with a new line 🙂

**Review by curquiza — CHANGES_REQUESTED** [review]
Thanks again for your PR, I request some small changes but the general work is really good 🙂

---

I don't know if you already know this on GitHub but here is my tip: you can accept the change suggestions directly on the GitHub interface here (this is an example):
Capture d’écran 2020-09-30 à 12 59 48

**Review by curquiza — APPROVED** [review]
Awesome @spahrson! Thanks for your reactivity!!

**Comment by curquiza** [comment]
bors merge

**Pull request merged**